base/common/LocalDateTime.h

```diff
@@ -67,7 +67,7 @@ class LocalDateTime
 
         m_hour = (s[11] - '0') * 10 + (s[12] - '0');
         m_minute = (s[14] - '0') * 10 + (s[15] - '0');
```

Comment:
Why we do not need such clamping for minutes, only for seconds? And why it shouldn't be checked in caller side? Such strings in input looks like invalid.

Reply:
You are right, it is required not only for seconds, here is an example:

```
$ clickhouse local -q "select toDateTime64(toUInt64(pow(2,33)), 1)" | cat -A
2062-10-10 ^@^@:56:32.0$
```

The reason is that DateLUTImpl cannot handle dates above 2106y, proper clamping was added in #20131

```
(gdb) p c.whole
$2 = 8589934592
(gdb) p date_lut.toHour(c.whole)                                                                                                         
$3 = 1572855
```